src/util.c

```diff
@@ -81,29 +81,18 @@ void checkMCFolder(void)
 
 static int checkMC()
 {
-    int dummy, ret, fd;
+    int fd;
 
     if (mcID == -1) {
-        mcGetInfo(0, 0, &dummy, &dummy, &dummy);
-        mcSync(0, NULL, &ret);
-
-        DIR *dir = opendir("mc0:OPL");
-        if (dir == NULL) {
-            dir = opendir("mc1:OPL");
-            if (dir == NULL) {
-                // No base dir found on any MC, check MC is inserted
-                fd = sysCheckMC();
-                if (fd == 0)
-                    mcID = 0x30;
-                else if (fd == 1)
-                    mcID = 0x31;
-            } else {
+        fd = sysCheckMC();
+        if (fd >= 0) {
+            DIR *dir = opendir("mc0:");
+            if (dir != NULL) {
                 closedir(dir);
-                mcID = 0x31;
+                mcID = 0x30;
             }
-        } else {
-            closedir(dir);
-            mcID = 0x30;
+            else
+                mcID = 0x31;
```

Comment:
I don't think you can assume "mc1:" exists if "mc0:" does not exist. Have you tested without any MC inserted?